src/terrain/clients/search.clj

```diff
@@ -1,13 +1,14 @@
 (ns terrain.clients.search
   (:use [terrain.util.config :only [search-base-url]]
-        [terrain.util.transformers :only [secured-params]])
+        [terrain.util.transformers :only [add-current-user-to-map]])
   (:require [clj-http.client :as http]
             [cemerick.url :refer [url]]))
 
 (defn do-data-search
   [body]
-  (let [req-options  {:form-params  body
-                      :query-params (secured-params)
+  (let [query-params {:user (or (:user (add-current-user-to-map {})) "anonymous")}
```

Comment:
I'd be a little tempted to enhance `add-current-user-to-map` to do something like this:

``` clojure
(defn add-current-user-to-map
  "Adds the name and e-mail address of the currently authenticated user to a
   map that can be used to generate a query string. If no user is authenticated,
   the `user` parameter is set to `anonymous` and the rest of the user attribute
   parameters are omitted."
  ([query default-username]
   (->> (assoc query
               :user       (or (:shortUsername current-user) default-username)
               :email      (:email current-user)
               :first-name (:firstName current-user)
               :last-name  (:lastName current-user))
        (remove-vals invalid-query-param-value?)))
  ([query]
   (add-current-user-to-map query "anonymous")))
```

Reply:
😄 This is a LOT cleaner - I'll refactor to this. Thanks!

Reply:
No problem at all. :)